framework/include/auxkernels/ElementL2IntegralAux.h

```diff
@@ -0,0 +1,52 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef ELEMENTL2INTEGRALAUX_H
+#define ELEMENTL2INTEGRALAUX_H
+
+// MOOSE includes
+#include "FunctionAux.h"
+
+// Forward declarations
+class ElementL2IntegralAux;
+
+template<>
+InputParameters validParams<ElementL2IntegralAux>();
+
+/**
+ * Compute an elemental field variable which is a single value per
+ * element equal to the L2-norm of a given function which is defined
+ * by derived classes.  The user is responsible for overriding the
+ * computeValue() function to return the quantity which should be
+ * squared, this class takes care of squaring it, doing the
+ * integration, and finally taking the square root.
+ */
+class ElementL2IntegralAux : public AuxKernel
+{
+public:
+  /**
+   * Class constructor
+   * @param name Object name
+   * @param parameters Object input parameters
+   */
+  ElementL2IntegralAux(const InputParameters & parameters);
+
+  /**
+   * Override the base class functionality to compute the element
+   * integral rather than scaling by element volume.
+   */
+  virtual void compute() override;
```

Comment:
> Where did you define compute()?

`compute()` is in ElementL2IntegralAux.C, I think you mean `computeValue()`?

That one is still pure virtual in this base class...  We could have have it take a "var", and then e.g. `ElementL2ErrorFunctionAux` would re-use this in place of what it currently calls `_soln`.